libzfs_core/_libzfs_core.py

```diff
@@ -0,0 +1,659 @@
+                       if that is required to receive the stream.
+    :param origin: the optional origin snapshot name if the stream is for a clone.
+    :type origin: bytes or None
+    :param props: the properties to set on the snapshot as *received* properties.
+    :type props: dict of bytes : Any
+
+    :raises IOError: if an input / output error occurs while reading from the ``fd``.
+    :raises DatasetExists: if the snapshot named ``snapname`` already exists.
+    :raises DatasetExists: if the stream is a full stream and the destination filesystem already exists.
+    :raises DatasetExists: if ``force`` is `True` but the destination filesystem could not
+                           be rolled back to a matching snapshot because a newer snapshot
+                           exists and it is an origin of a cloned filesystem.
+    :raises StreamMismatch: if an incremental stream is received and the latest
+                            snapshot of the destination filesystem does not match
+                            the source snapshot of the stream.
+    :raises StreamMismatch: if a full stream is received and the destination
+                            filesystem already exists and it has at least one snapshot,
+                            and ``force`` is `False`.
+    :raises StreamMismatch: if an incremental clone stream is received but the specified
+                            ``origin`` is not the actual received origin.
+    :raises DestinationModified: if an incremental stream is received and the destination
+                                 filesystem has been modified since the last snapshot
+                                 and ``force`` is `False`.
+    :raises DestinationModified: if a full stream is received and the destination
+                                 filesystem already exists and it does not have any
+                                 snapshots, and ``force`` is `False`.
+    :raises DatasetNotFound: if the destination filesystem and its parent do not exist.
+    :raises DatasetNotFound: if the ``origin`` is not `None` and does not exists.
+    :raises DatasetBusy: if ``force`` is `True` but the destination filesystem could not
+                         be rolled back to a matching snapshot because a newer snapshot
+                         is held and could not be destroyed.
+    :raises DatasetBusy: if another receive operation is being performed on the
+                         destination filesystem and this operation has lost the race.
+    :raises BadStream: if the stream is corrupt or it is not recognized or it is
+                       a compound stream or it is a clone stream, but ``origin``
+                       is `None`.
+    :raises BadStream: if a clone stream is received and the destination filesystem
+                       already exists.
+    :raises StreamFeatureNotSupported: if the stream has a feature that is not
+                                       supported on the receiving side.
```

Comment:
The receiving side is "this" side - ie, where this code is running - always, right?